polkadot/xcm/xcm-executor/src/lib.rs

```diff
@@ -630,9 +630,39 @@ impl<Config: config::Config> XcmExecutor<Config> {
 				Ok(())
 			},
 			DepositReserveAsset { assets, dest, xcm } => {
-				let deposited = self.holding.saturating_take(assets);
-				for asset in deposited.assets_iter() {
-					Config::AssetTransactor::deposit_asset(&asset, &dest, Some(&self.context))?;
+				let old_holding = self.holding.clone();
+				let result = Config::TransactionalProcessor::process(|| {
```

Comment:
This adds both transactionality and delivery-fees-fix to `DepositReserveAsset`, shouldn't we also backport transactionality for all others too?

Reply:
Yes, cherry picked the other commit